### PR TITLE
[RAPTOR-3769] find nested custom files for R fit

### PR DESCRIPTION
**Pull request by jlpolit**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

### CODE REVIEW

File: custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R [hunk 1]
```diff
@@ -7,13 +7,26 @@ fit_hook <- FALSE
...
-  if (file.exists(file.path(code_dir, 'custom.R'))){
-    custom_path <- file.path(code_dir, "custom.R")
-  } else {
-    custom_path <- file.path(code_dir, "custom.r")
+str_right <- function(string, n) {
+  substr(string, nchar(string) - (n - 1), nchar(string))
+}
+
+get_nested_path <- function(path, file_pattern){
+  if(str_right(path, 1) == '/'){
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R:15]: please put space after if

File: custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R [hunk 2]
```diff
@@ -7,13 +7,26 @@ fit_hook <- FALSE
...
   }
-  custom_loaded <- import(custom_path)
+  return(
+    list.files(path=path, pattern=file_pattern, recursive = TRUE, full.names=TRUE)
+  )
+}
+
+init <- function(code_dir) {
+    custom_path <- get_nested_path(code_dir, 'custom.[Rr]')
+    if(length(custom_path) >= 1){
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R:25]: and here

File: custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R [hunk 3]
```diff
@@ -7,13 +7,26 @@ fit_hook <- FALSE
...
+        custom_loaded <- import(custom_path[1])
+    } else{
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/resource/components/Python/r_fit/fit.R:27]: and here

**Review by yakov-g — APPROVED** [review]

**Review by meltzg — APPROVED** [review]

**Comment by datarobot-jarvis** [comment]
:dollar: Total cost of executing tests in Jenkins: $**1.17** USD.

[What is this all about?](http://bit.ly/2BahK4V)

**Pull request merged**